notebooks/create_similarity_matrix.py

```diff
@@ -0,0 +1,366 @@
+import os
+import pathos.multiprocessing
+import sys
+from pathlib import Path
+
+# Make sure all code is in the PATH.
+sys.path.append("../src/")
+
+import numba as nb
+import numpy as np
+import pandas as pd
+import pyteomics.mgf
+import tqdm
+import spectrum_utils.spectrum as sus
+import similarity
+import bile_mods
+
+# from multiprocessing import Pool, freeze_support
+# from multiprocessing import freeze_support
+from functools import partial
+from pathos.multiprocessing import ProcessingPool as Pool
+
+# public parameters
+library_file = "../data/BILELIB19.mgf"
+# library_file = "../data/20220418_ALL_GNPS_NO_PROPOGATED.mgf"
+
+# analysis name
+analysis_name = "as_exchange"
+
+# square root transformation of intensities is often performed to limit the impact of high abundant signals
+apply_sqrt = True
+
+# size of subset of spectral pairs
+n_spectral_pairs = 500000
+
+# minimum number of signals only removes the spectra with less
+min_n_signals = 6
+
+# signal alignment tolerance
+abs_mz_tolerance = 0.02
+# only allow precursor mz difference of:
+max_mz_delta = 200
+# if defined, we will only search for specific delta m/z between two spectra
+# 16 oxygen 15.994914
+# otherwise define as -1
+# specific_mod_mz = 15.9949
+specific_mod_mz = -1
+
+# free bile acids compared to conjugated with amino acids
+# or different delta mz between different conjugated bile acids
+# mod_list = np.empty(0, float)
+# mod_list = bile_mods.get_as_mods()
+mod_list = bile_mods.get_as_exchange()
+
+library_file_name_without_ext = Path(library_file).stem
+# analysis ID is used for file export
+if specific_mod_mz <= 0:
+    analysis_id = "{}_{}_sqrt_{}_{}pairs_{}min_signals_{}maxdelta_{}mods".format(
+        library_file_name_without_ext,
+        analysis_name,
+        apply_sqrt,
+        n_spectral_pairs,
+        min_n_signals,
+        max_mz_delta,
+        len(mod_list),
+    ).replace(".", "i")
+else:
+    analysis_id = "{}_{}_sqrt_{}_{}pairs_{}min_signals_{}specific_delta_{}mods".format(
+        library_file_name_without_ext,
+        analysis_name,
+        apply_sqrt,
+        n_spectral_pairs,
+        min_n_signals,
+        specific_mod_mz,
+        len(mod_list),
+    ).replace(".", "i")
+
+# output filename of pairs only with pair selection relevant parameters:
+pairs_filename = (
+    "temp/{}_pairs.parquet".format(analysis_id)
+    .replace("sqrt_True_", "")
+    .replace("sqrt_False_", "")
+)
+spectra_filename = "tempspectra/spectra_{}_{}min_signals_sqrt_{}.parquet".format(
+    library_file_name_without_ext, min_n_signals, apply_sqrt
+)
+
+
+def main():
+    spectra = None
+    precursor_mz_list = None
+    if (os.path.isfile(pairs_filename) == False) or (
```

Comment:
A bit weird that you explicitly specify `== False`, the Pythonic way would instead be `not ...`.

Reply:
I know I changed it. In the beginning the whole thing was more complex and after simplifying I did not change it again